--- a/cobbler/api.py	2019-09-04 14:23:53.000000000 +0300
+++ b/cobbler/api.py	2019-11-21 23:03:56.140304470 +0300
@@ -857,8 +857,8 @@
                 except:
                     self.log("Network root given to --available-as is missing a colon, please see the manpage example.")
                     return False
-
-        import_module = self.get_module_by_name("manage_import_signatures").get_import_manager(self._collection_mgr, logger)
+        self.log(repr(module_loader.MODULE_CACHE.keys()))
+        import_module = self.get_module_by_name("managers.import_signatures").get_import_manager(self._collection_mgr, logger)
         import_module.run(path, mirror_name, network_root, autoinstall_file, arch, breed, os_version)
 
     # ==========================================================================
