From d198c9ca7b3475abdbb5a369f018645e43980021 Mon Sep 17 00:00:00 2001
From: Alexander Miroshnichenko <a.miroshnichenko@rbk.money>
Date: Tue, 29 Oct 2019 15:09:25 +0300
Subject: [PATCH] file.get_selinux_context: fix directory path behaviour
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Signed-off-by: Alexander Miroshnichenko <a.miroshnichenko@rbk.money>
---
 salt/modules/file.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/salt/modules/file.py b/salt/modules/file.py
index d0847c669b76..03fedbc6c371 100644
--- a/salt/modules/file.py
+++ b/salt/modules/file.py
@@ -3850,7 +3850,7 @@ def get_selinux_context(path):
 
         salt '*' file.get_selinux_context /etc/hosts
     '''
-    out = __salt__['cmd.run'](['ls', '-Z', path], python_shell=False)
+    out = __salt__['cmd.run'](['ls', '-dZ', path], python_shell=False)
 
     try:
         ret = re.search(r'\w+:\w+:\w+:\w+', out).group(0)
-- 
2.23.0

