--- a/plugins/check_mysql.c	2015-10-16 12:06:18.000000000 +0300
+++ b/plugins/check_mysql.c	2016-02-20 17:17:48.277328058 +0300
@@ -50,6 +50,7 @@
 char *db_socket = NULL;
 char *db_pass = NULL;
 char *db = NULL;
+char *slave_connection_name = NULL;
 char *ca_cert = NULL;
 char *ca_dir = NULL;
 char *cert = NULL;
@@ -109,7 +110,8 @@
 	char *result = NULL;
 	char *error = NULL;
 	char slaveresult[SLAVERESULTSIZE];
-	char* perf;
+	char *query;
+	char *perf;
 
         perf = strdup ("");
 
@@ -206,12 +208,19 @@
 	}
 
 	if(check_slave) {
-		/* check the slave status */
-		if (mysql_query (&mysql, "show slave status") != 0) {
-			error = strdup(mysql_error(&mysql));
-			mysql_close (&mysql);
-			die (STATE_CRITICAL, _("slave query error: %s\n"), error);
-		}
+	  /* build query */
+	  if (strcmp(slave_connection_name, "") != 0) {
+	    asprintf(&query, "show slave '%s' status", slave_connection_name);
+	  } else {
+	    asprintf(&query, "show slave status");
+	  }
+	  /* check the slave status */
+	  if (mysql_query (&mysql, query) != 0) {
+	    error = strdup(mysql_error(&mysql));
+	    mysql_close (&mysql);
+	    die (STATE_CRITICAL, _("slave query error: %s\n"), error);
+	  }
+	  free(query);
 
 		/* store the result */
 		if ( (res = mysql_store_result (&mysql)) == NULL) {
@@ -352,6 +361,7 @@
 		{"critical", required_argument, 0, 'c'},
 		{"warning", required_argument, 0, 'w'},
 		{"check-slave", no_argument, 0, 'S'},
+		{"slave-connection-name", no_argument, 0, 'N'},
 		{"ignore-auth", no_argument, 0, 'n'},
 		{"verbose", no_argument, 0, 'v'},
 		{"version", no_argument, 0, 'V'},
@@ -369,7 +379,7 @@
 		return ERROR;
 
 	while (1) {
-		c = getopt_long (argc, argv, "hlvVnSP:p:u:d:H:s:c:w:a:k:C:D:L:f:g:", longopts, &option);
+		c = getopt_long (argc, argv, "hlvVnSP:p:u:d:H:s:c:w:a:k:C:D:L:f:g:N:", longopts, &option);
 
 		if (c == -1 || c == EOF)
 			break;
@@ -431,6 +441,9 @@
 		case 'S':
 			check_slave = 1;							/* check-slave */
 			break;
+		case 'N':                                                                       /* slave-connection-name */
+		  slave_connection_name = optarg;
+		  break;
 		case 'n':
 			ignore_auth = 1;							/* ignore-auth */
 			break;
@@ -497,6 +510,9 @@
 	if (db == NULL)
 		db = strdup("");
 
+	if (slave_connection_name == NULL)
+	  slave_connection_name = strdup("");
+
 	return OK;
 }
 
@@ -541,6 +557,8 @@
   printf ("    %s\n", _("Your clear-text password could be visible as a process table entry"));
   printf (" %s\n", "-S, --check-slave");
   printf ("    %s\n", _("Check if the slave thread is running properly."));
+  printf (" %s\n", "-N, --slave-connection-name");
+  printf ("    %s\n", _("Connection name if using multi-source replication."));
   printf (" %s\n", "-w, --warning");
   printf ("    %s\n", _("Exit with WARNING status if slave server is more than INTEGER seconds"));
   printf ("    %s\n", _("behind master"));
