diff -ur systemd-stable-245.5/src/basic/missing_type.h systemd-stable-245.5.new/src/basic/missing_type.h
--- systemd-stable-245.5/src/basic/missing_type.h	2020-04-17 12:37:12.000000000 +0000
+++ systemd-stable-245.5.new/src/basic/missing_type.h	2020-06-30 12:13:34.326587133 +0000
@@ -10,3 +10,23 @@
 #if !HAVE_CHAR16_T
 #define char16_t uint16_t
 #endif
+
+#ifndef FTW_ACTIONRETVAL
+#define FTW_ACTIONRETVAL 16
+#endif
+
+#ifndef FTW_CONTINUE
+#define FTW_CONTINUE 0
+#endif
+
+#ifndef FTW_STOP
+#define FTW_STOP 1
+#endif
+
+#ifndef FTW_SKIP_SUBTREE
+#define FTW_SKIP_SUBTREE 2
+#endif
+
+#ifndef FTW_SKIP_SIBLINGS
+#define FTW_SKIP_SIBLINGS 3
+#endif
Only in systemd-stable-245.5.new/src/basic: missing_type.h.orig
diff -ur systemd-stable-245.5/src/core/kmod-setup.c systemd-stable-245.5.new/src/core/kmod-setup.c
--- systemd-stable-245.5/src/core/kmod-setup.c	2020-04-17 12:37:12.000000000 +0000
+++ systemd-stable-245.5.new/src/core/kmod-setup.c	2020-06-30 12:16:43.286587133 +0000
@@ -9,6 +9,7 @@
 #include "fileio.h"
 #include "kmod-setup.h"
 #include "macro.h"
+#include "missing_type.h"
 #include "string-util.h"
 
 #if HAVE_KMOD
diff -ur systemd-stable-245.5/src/core/mount-setup.c systemd-stable-245.5.new/src/core/mount-setup.c
--- systemd-stable-245.5/src/core/mount-setup.c	2020-04-17 12:37:12.000000000 +0000
+++ systemd-stable-245.5.new/src/core/mount-setup.c	2020-06-30 12:15:46.596587133 +0000
@@ -21,6 +21,7 @@
 #include "label.h"
 #include "log.h"
 #include "macro.h"
+#include "missing_type.h"
 #include "mkdir.h"
 #include "mount-setup.h"
 #include "mountpoint-util.h"
